builtin/logical/postgresql/path_role_create.go

```diff
@@ -77,40 +81,52 @@ func (b *backend) pathRoleCreateRead(
 		Add(lease.Lease).
 		Format("2006-01-02 15:04:05-0700")
 
-	// Get our connection
+	// Get our handle
+	b.logger.Println("[TRACE] postgres/pathRoleCreateRead: getting database handle")
 	db, err := b.DB(req.Storage)
 	if err != nil {
 		return nil, err
 	}
 
 	// Start a transaction
+	b.logger.Println("[TRACE] postgres/pathRoleCreateRead: starting transaction")
 	tx, err := db.Begin()
 	if err != nil {
 		return nil, err
 	}
-	defer tx.Rollback()
+	defer func() {
+		b.logger.Println("[TRACE] postgres/pathRoleCreateRead: rolling back transaction")
+		tx.Rollback()
+	}()
 
 	// Execute each query
 	for _, query := range SplitSQL(role.SQL) {
-		stmt, err := db.Prepare(Query(query, map[string]string{
+		b.logger.Println("[TRACE] postgres/pathRoleCreateRead: preparing statement")
+		stmt, err := tx.Prepare(Query(query, map[string]string{
 			"name":       username,
 			"password":   password,
 			"expiration": expiration,
 		}))
 		if err != nil {
 			return nil, err
 		}
+		defer stmt.Close()
```

Comment:
Has anyone looked into the efficiency/performance of the way the database connection pooling happens in Go w.r.t. PostgreSQL's `PREPARE`?  If we're not reusing the `PREPARE`'ed statement and are re-preparing every request, it's going to probably be significantly faster to just `Exec`.  If, however, we do see `PREPARE`'ed statements cached and reused, then maintaining the `PREPARE` is useful.  I don't know if the Go Db driver is doing anything fancy under the hood to make this magic or not. Also CC @LLBennett